cmd/kube-scheduler/app/options/globalflags.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"fmt"
+
+	"github.com/spf13/pflag"
+
+	// libs that provide registration functions
+	"k8s.io/apiserver/pkg/util/flag"
+	"k8s.io/apiserver/pkg/util/logs"
+	"k8s.io/kubernetes/pkg/version/verflag"
+)
+
+// AddGlobalFlags explicitly registers flags that libraries (glog, verflag, etc.) register
+// against the global flagsets from "flag" and "github.com/spf13/pflag".
+// We do this in order to prevent unwanted flags from leaking into the kube-scheduler's flagset.
+func AddGlobalFlags(fs *pflag.FlagSet, name string) {
+	flag.AddGlogFlags(fs)
+	verflag.AddFlags(fs)
+	logs.AddFlags(fs)
+
```

Comment:
If i move the `AddGlobalFlags()` into `k8s.io/apiserver/pkg/util/flag/glog_flag.go` i created in this PR, there have panic like that when i `local-up` cluster in local:
```
goroutine 1 [running]:
k8s.io/kubernetes/vendor/github.com/spf13/pflag.(*FlagSet).AddFlag(0xc0000c0000, 0xc000159720)
	/ycj/kubernetes-community/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/pflag/flag.go:836 +0x870
k8s.io/kubernetes/vendor/github.com/spf13/pflag.(*FlagSet).VarPF(0xc0000c0000, 0x19e1040, 0xc00003e760, 0x17ec83d, 0x13, 0x0, 0x0, 0x1811409, 0x2d, 0x7f4810d886c0)
	/ycj/kubernetes-community/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/pflag/flag.go:819 +0x10b
k8s.io/kubernetes/vendor/github.com/spf13/pflag.(*FlagSet).VarP(0xc0000c0000, 0x19e1040, 0xc00003e760, 0x17ec83d, 0x13, 0x0, 0x0, 0x1811409, 0x2d)
	/ycj/kubernetes-community/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/pflag/flag.go:825 +0x8e
k8s.io/kubernetes/vendor/github.com/spf13/pflag.(*FlagSet).DurationVarP(0xc0000c0000, 0xc00003e760, 0x17ec83d, 0x13, 0x0, 0x0, 0x12a05f200, 0x1811409, 0x2d)
	/ycj/kubernetes-community/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/pflag/duration.go:48 +0x95
k8s.io/kubernetes/vendor/github.com/spf13/pflag.(*FlagSet).DurationP(0xc0000c0000, 0x17ec83d, 0x13, 0x0, 0x0, 0x12a05f200, 0x1811409, 0x2d, 0xc0001e1e60)
	/ycj/kubernetes-community/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/pflag/duration.go:73 +0x9b
k8s.io/kubernetes/vendor/github.com/spf13/pflag.Duration(0x17ec83d, 0x13, 0x12a05f200, 0x1811409, 0x2d, 0xc0004bdbc0)
	/ycj/kubernetes-community/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/pflag/duration.go:80 +0x67
hack/../cluster/../cluster/../_output/bin/kubectl flag redefined: log-flush-frequency
panic: hack/../cluster/../cluster/../_output/bin/kubectl flag redefined: log-flush-frequency
```
But if i remove `logs.AddFlags(fs)` and move function again, the panic gone, have any idea about it?  @sttts @liggitt

Reply:
it looks like we have two packages that register that flag in package variables:

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/util/logs/logs.go#L29

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/util/logs/logs.go#L32

I assume moving those around introduced a transitive dependency that included both packages at once.